packages/atomic/src/components/atomic-folded-result-list/atomic-folded-result-list.tsx

```diff
@@ -0,0 +1,316 @@
+  @InitializeBindings() public bindings!: Bindings;
+  private resultList!: FoldedResultList;
+  public resultsPerPage!: ResultsPerPage;
+  private resultTemplatesManager!: ResultTemplatesManager<TemplateContent>;
+
+  @Element() private host!: HTMLDivElement;
+
+  @BindStateToController('resultList')
+  @State()
+  private resultListState!: FoldedResultListState;
+
+  @BindStateToController('resultsPerPage')
+  @State()
+  private resultsPerPageState!: ResultsPerPageState;
+
+  @State() public error!: Error;
+  @State() private templateHasError = false;
+
+  /**
+   * TODO:
+   */
+  @Prop({reflect: true}) public foldingCollection?: string;
+  /**
+   * TODO:
+   */
+  @Prop({reflect: true}) public foldingParent?: string;
+  /**
+   * TODO:
+   */
+  @Prop({reflect: true}) public foldingChild?: string;
+
+  /**
+   * TODO: KIT-452 Infinite scroll feature
+   * Whether to automatically retrieve an additional page of results and append it to the
+   * current results when the user scrolls down to the bottom of element
+   */
+  private enableInfiniteScroll = false;
+  /**
+   * A list of non-default fields to include in the query results, separated by commas.
+   * The default fields sent in a request are: 'date', 'author', 'source', 'language', 'filetype', 'parents', ‘urihash’, ‘objecttype’, ‘collection’, ‘permanentid’ 'ec_price', 'ec_name', 'ec_description', 'ec_brand', 'ec_category', 'ec_item_group_id', 'ec_shortdesc', 'ec_thumbnails', 'ec_images', 'ec_promo_price', 'ec_in_stock', 'ec_cogs', and 'ec_rating'.
+   */
+  @Prop({reflect: true}) public fieldsToInclude = '';
+  /**
+   * The spacing of various elements in the result list, including the gap between results, the gap between parts of a result, and the font sizes of different parts in a result.
+   */
+  @Prop({reflect: true}) density: ResultDisplayDensity = 'normal';
+  /**
+   * The expected size of the image displayed in the results.
+   */
+  @Prop({reflect: true}) imageSize?: ResultDisplayImageSize;
+  /**
+   * @deprecated use `imageSize` instead.
+   */
+  @Prop({reflect: true}) image: ResultDisplayImageSize = 'icon';
+
+  private renderingFunction?: (result: FoldedResult) => HTMLElement = undefined;
+
+  /**
+   * Sets a rendering function to bypass the standard HTML template mechanism for rendering results.
+   * You can use this function while working with web frameworks that don't use plain HTML syntax, e.g., React, Angular or Vue.
+   *
+   * Do not use this method if you integrate Atomic in a plain HTML deployment.
+   *
+   * @param render
+   */
+  @Method() public async setRenderFunction(
+    render: (result: FoldedResult) => HTMLElement
+  ) {
+    this.renderingFunction = render;
+  }
+
+  private listWrapperRef?: HTMLDivElement;
+
+  private get fields() {
+    if (this.fieldsToInclude.trim() === '') return [];
+    return this.fieldsToInclude.split(',').map((field) => field.trim());
+  }
+
+  private get defaultFieldsToInclude() {
```

Comment:
This is no directly related to this PR, but this list should probably be defined and exported from Headless (and then used as source of truth), instead of duplicating it in many place in the codebase.

Anyway, it's fine to leave like this for now !